bundles/ch.elexis.laborimport_labtop/src/ch/elexis/laborimport/labtop/Importer.java

```diff
@@ -196,7 +196,31 @@ public IStatus doImport(final IProgressMonitor monitor) throws Exception {
 			String filename = results[1];
 			return ResultAdapter.getResultAsStatus(hlp.importFile(filename, false));
 		} else {
-			return ResultAdapter.getResultAsStatus(importDirect());
+			if (openmedicalObject != null) {
+				return ResultAdapter.getResultAsStatus(importDirect());
+			} else {
+				File downloadPath = new File(CoreHub.localCfg.get(PreferencePage.DL_DIR, null));
+				File archiveDir = new File(downloadPath, "archive"); //$NON-NLS-1$
+				String[] files = downloadPath.list(new FilenameFilter() {
+					@Override
+					public boolean accept(File path, String name) {
+						if (name.toLowerCase().endsWith(".hl7")) { //$NON-NLS-1$
+							return true;
+						}
+						return false;
+					}
+				});
+				Result<?> rs = null;
+				for (String file : files) {
+					File f = new File(downloadPath, file);
+					try {
+						rs = hlp.importFile(f, archiveDir, false);
+					} catch (IOException e) {
+						SWTHelper.showError("Import error", e.getMessage()); //$NON-NLS-1$
+					}
+				}
+				return ResultAdapter.getResultAsStatus(rs);
```

Comment:
Sieht für mich so aus als ob dieser code auch in der `importDirect` Methode vor kommt. Bitte das in einer separaten `importDirectory` Methode 1 mal impl. und dann hier und in `importDirect` diese Methode verwenden.